datafusion/proto/src/bytes/mod.rs

```diff
@@ -107,6 +107,19 @@ pub fn logical_plan_to_bytes(plan: &LogicalPlan) -> Result<Bytes> {
     logical_plan_to_bytes_with_extension_codec(plan, &extension_codec)
 }
 
+/// Serialize a LogicalPlan as json
+#[cfg(feature = "json")]
```

Comment:
The only weird part about this PR is that these methods just disappear if the `json` feature is disabled - would it be better to put them in another file or something?

Reply:
I see no issue with this, it is fairly normal across the ecosystem